### PR TITLE
Make handshake dependencies optional.

### PR DESCRIPTION
**Pull request by Icelk**
This is the PR I promised in snapview/tungstenite-rs#296.
It requires snapview/tungstenite-rs#300 to be merged and released before we can release this.

This adds a cargo feature named `handshake` which is enabled by default. Disabling it removes the handshake feature of tungstenite.

### CODE REVIEW

**Comment by Icelk** [comment]
I can confirm this works as expected. I disabled the handshake feature and handled that myself.

**Comment by Icelk** [comment]
The CI is failing because I added `tungstenite` as a path dependency for development. Should I add it as a git reference, so the CI succeeds?

**Comment by application-developer-DA** [comment]
Thanks for the changes! It's ok, let's wait until we merge the `tungstenite` and release it, so that we can merge this one and release it as well.

File: Cargo.toml [hunk 1]
```diff
@@ -32,6 +33,7 @@ tokio = { version = "1.0.0", default-features = false, features = ["io-util"] }
 
 [dependencies.tungstenite]
 version = "0.17.3"
+path = "../tungstenite-rs"
```
Comment (daniel-abramov) [inline Cargo.toml:36]:
```suggestion
version = "0.18.0"
```

**Review by daniel-abramov — COMMENTED** [review]
Now we can update and merge this :)

**Comment by Icelk** [comment]
Great! Thanks!

**Comment by Icelk** [comment]
Just a reminder: should the docs in `README` be updated?

**Comment by daniel-abramov** [comment]
Possibly. We did not document the `connect` feature though. But perhaps it makes sense to document both `connect` and `handshake`.

**Pull request merged**